### ISSUE TITLE
search on string returns Sql error E42601

### ISSUE THREAD

**Issue by konotorii** [prisma/prisma#18145]
### Bug description

So basically whenever I try to use the ``search: {}`` on a string I get this error:

```
ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E42601), message: "syntax error in tsquery: \"the string im searching\"", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("tsquery.c"), line: Some(689), routine: Some("makepol") }) }), transient: false })
```

This has been happening since version 4.5 or so.

### How to reproduce

Not sure if this can be reproduced....

### Expected behavior

Being able to use ``search: {}`` without any errors and getting a result.

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  binaryTargets   = ["native", "linux-musl"]
  previewFeatures = ["fullTextSearch", "fieldReference", "tracing", "metrics"]
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model user {
  id Int @id @unique @default(autoincrement())
  username String
}
```

```ts
const user = await this.prisma.user.findFirst({
            where: {
                OR: [
                    {
                        id: Number(id) || 0
                    },
                    {
                        username: {
                            search: String(id)
                        }
                    }
                ]
            }
        });
```

### Environment & setup

- OS: Ubuntu 22.04
- Database: Postgres 14.5
- Node.js version: 18.14.2

### Prisma Version

```
prisma                  : 4.10.1
@prisma/client          : 4.10.1
Current platform        : linux-musl-openssl-3.0.x
Query Engine (Node-API) : libquery-engine aead147aa326ccb985dcfed5b065b4fdabd44b19 (at node_modules/@prisma/engines/libquery_engine-linux-musl-openssl-3.0.x.so.node)
Migration Engine        : migration-engine-cli aead147aa326ccb985dcfed5b065b4fdabd44b19 (at node_modules/@prisma/engines/migration-engine-linux-musl-openssl-3.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.10.1-1.80b351cc7c06d352abe81be19b8a89e9c6b7c110
Default Engines Hash    : aead147aa326ccb985dcfed5b065b4fdabd44b19
Studio                  : 0.481.0
Preview Features        : fullTextSearch, tracing, metrics, fieldReference
```

**Comment by cabjr** [comment]
I'm having the same issue, In my case, it is happening with any search using the orderBy _relevance, my stack trace is:

clientVersion: 4.8.0

`\nInvalid `this.prisma.job.findMany()` invocation in\n/coderepo/myclass.repository.js:34:54\n\n  31 }\n  32 findMany(params) {\n  33     return __awaiter(this, void 0, void 0, function* () {\n→ 34         const results= yield this.prisma.myentity.findMany(\nError occurred during query execution:\nConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: \"ERROR\", parsed_severity: Some(Error), code: SqlState(E42601), message: \"syntax error in tsquery: \\\"software developer\\\"\", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: Some(\"tsquery.c\"), line: Some(726), routine: Some(\"makepol\") }) }) })`



```
prisma.myentity.findmany({
where: {...}, 
orderBy: [
  {
    _relevance: {
      fields: ['title', 'description'],
      search: 'test',
      sort: 'asc',
    }
  },
})
```

**Comment by araujodavid** [comment]
I'm getting the same error with Prisma `4.13.0`.

**Comment by konotorii** [comment]
i was able to fix this by adding ``.toLowerCase()`` to my string

**Comment by konotorii** [comment]
further testing shows that spaces become the problem, any reason as to why @ Prisma Team?

**Comment by Samoboolean** [comment]
I'm also getting this error when I'm searching for text with "!". Is there an easy to parse a user's input to removes potential query operators? Or perhaps just have a 'raw' search mode that ignores query operators.

**Comment by konotorii** [comment]
from what ive seen it appears that spaces, uppercase letters, maybe numbers (need to test) and special characters cause this error

**Comment by devSajan** [comment]
I am facing the same type of error with a different error code. the log format is flatted stringify format.

ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E40P01), message: "deadlock detected", detail: Some("Process 7627 waits for ShareLock on transaction 920972; blocked by process 6362.\nProcess 6362 waits for ShareLock on transaction 921008; blocked by process 6359.\nProcess 6359 waits for ExclusiveLock on tuple (3,48) of relation 1191426 of database 14717; blocked by process 7627."), hint: Some("See server log for query details."), position: None, where_: Some("while updating tuple (3,48) in relation \"user_fiat_balances\""), schema: None, table: None, column: None, datatype: None, constraint: None, file: Some("deadlock.c"), line: Some(1152), routine: Some("DeadLockReport") }) }), transient: false })

**Comment by FGoessler** [comment]
This seems to be a duplicate of https://github.com/prisma/prisma/issues/24042
